pkg/uploadserver/uploadserver_test.go

```diff
@@ -204,209 +200,175 @@ func replaceAsyncProcessorFunc(replacement func(io.ReadCloser, string, string, s
 	}()
 	f()
 }
-func TestGetFails(t *testing.T) {
-	withProcessorSuccess(func() {
-		req, err := http.NewRequest("GET", common.UploadPathSync, nil)
-		if err != nil {
-			t.Fatal(err)
-		}
 
-		rr := httptest.NewRecorder()
+var _ = Describe("Upload server tests", func() {
+	It("GET fails", func() {
+		withProcessorSuccess(func() {
+			req, err := http.NewRequest("GET", common.UploadPathSync, nil)
+			Expect(err).ToNot(HaveOccurred())
 
-		server := newServer()
-		server.ServeHTTP(rr, req)
+			rr := httptest.NewRecorder()
 
-		if status := rr.Code; status != http.StatusNotFound {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusNotFound)
-		}
+			server := newServer()
+			server.ServeHTTP(rr, req)
+
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusNotFound))
+		})
 	})
-}
 
-func TestHealthz(t *testing.T) {
-	req, err := http.NewRequest("GET", healthzPath, nil)
-	if err != nil {
-		t.Fatal(err)
-	}
+	It("healthz", func() {
+		req, err := http.NewRequest("GET", healthzPath, nil)
+		Expect(err).ToNot(HaveOccurred())
 
-	rr := httptest.NewRecorder()
+		rr := httptest.NewRecorder()
 
-	app := uploadServerApp{}
-	server, _ := app.createHealthzServer()
-	server.Handler.ServeHTTP(rr, req)
+		app := uploadServerApp{}
+		server, _ := app.createHealthzServer()
+		server.Handler.ServeHTTP(rr, req)
 
-	if status := rr.Code; status != http.StatusOK {
-		t.Errorf("handler returned wrong status code: got %v want %v",
-			status, http.StatusOK)
-	}
-}
+		status := rr.Code
+		Expect(status).To(Equal(http.StatusOK))
 
-func TestInProcessUnavailable(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newRequest(t)
+	})
 
-		rr := httptest.NewRecorder()
+	It("Process unavailable", func() {
+		withProcessorSuccess(func() {
+			req := newRequest()
+
+			rr := httptest.NewRecorder()
 
-		server := newServer()
-		server.uploading = true
-		server.ServeHTTP(rr, req)
+			server := newServer()
+			server.uploading = true
+			server.ServeHTTP(rr, req)
+
+			status := rr.Code
+			Expect(status).To(Equal(http.StatusServiceUnavailable))
+		})
 
-		if status := rr.Code; status != http.StatusServiceUnavailable {
-			t.Errorf("handler returned wrong status code: got %v want %v",
-				status, http.StatusServiceUnavailable)
-		}
 	})
-}
 
-func TestInProcessUnavailableAsync(t *testing.T) {
-	withProcessorSuccess(func() {
-		req := newAsyncRequest(t)
+	It("Process unavailable, async", func() {
```

Comment:
Only difference between this one and the one above is the request type, can use a table with the request as a parameter to reduce code.